src/librustdoc/html/static/js/search.js

```diff
@@ -111,6 +112,157 @@ function levenshtein(s1, s2) {
     return s1_len + s2_len;
 }
 
+function extractInformationFromElem(container, className) {
+    return Array.from(container.querySelectorAll(`.${className} .argument`)).map(elem => {
+        const text = elem.querySelector(".arg").value.trim();
+        const generics = elem.querySelector(".generic").value.trim();
+
+        if (generics.length !== 0) {
+            return `${text}<${generics}>`;
+        }
+        return text;
+    }).filter(text => text.length !== 0);
+}
+
+function generateSearch(literalSearch) {
+    const container = document.getElementById("search-helper");
+    const filter = container.querySelector("select").value;
+    const args = extractInformationFromElem(container, "search-arguments");
+    const returned = extractInformationFromElem(container, "search-returned");
+
+    let query = "";
+
+    if (args.length === 0 && returned.length === 0) {
+        return;
+    }
+
+    if (filter.length !== 0) {
+        query += `${filter}:`;
+    }
+
+    if (literalSearch) {
+        if (args.length !== 0) {
+            query += `"${args[0]}"`;
+        } else {
+            if (query.length !== 0) {
+                query += " ";
+            }
+            query += `-> "${returned[0]}"`;
+        }
+    } else {
+        query += args.join(", ");
+        if (returned.length !== 0) {
+            if (query.length !== 0) {
+                query += " ";
+            }
+            query += `-> ${returned.join(", ")}`;
+        }
+    }
+    document.querySelector(".search-container .search-input").value = query;
+    // Now collapsing the advanced search.
+    container.removeAttribute("open");
+}
+
+function addAdvancedSearchUI() {
+    const filters = itemTypesMatching.map(elem => `<option value=${elem[1]}>${elem[0]}</option>`);
+    const detailsElem = document.createElement("details");
+    let literalSearch = false;
+
+    detailsElem.id = "search-helper";
+    detailsElem.innerHTML = `\
+<summary>Advanced search helper</summary>
+<form>
+    <select>
+        <option value="">&lt;no filter&gt;</option>
+    ${filters.join("")}
+    </select>
+    <div class="search-literal">
+        <input type="checkbox">
+        <label>Enable literal search</label>
+    </div>
+    <div class="search-arguments">
+        <label>Arguments</label>
+        <div class="argument">
+            <input placeholder="function/method arguments" class="arg">
```

Comment:
I think "function" covers both.

```suggestion
            <input placeholder="function arguments" class="arg">
```